eclair-core/src/main/scala/fr/acinq/eclair/crypto/ChaCha20Poly1305.scala

```diff
@@ -106,7 +113,12 @@ object ChaCha20Poly1305 extends Logging {
     val polykey = ChaCha20.encrypt(ByteVector32.Zeroes, key, nonce)
     val ciphertext = ChaCha20.encrypt(plaintext, key, nonce, 1)
     val tag = Poly1305.mac(polykey, aad, pad16(aad), ciphertext, pad16(ciphertext), Protocol.writeUInt64(aad.length, ByteOrder.LITTLE_ENDIAN), Protocol.writeUInt64(ciphertext.length, ByteOrder.LITTLE_ENDIAN))
+
     logger.debug(s"encrypt($key, $nonce, $aad, $plaintext) = ($ciphertext, $tag)")
+    if (nonce === ByteVector.fill(ChaCha20.NonceLength)(0.byteValue)) {
```

Comment:
I'd rather compare nonce to a `Zeroes` val, with a comment that explains that nonce is incremental and initialised with 0.

Reply:
Sounds good, done.